pkg/dbmate/db.go

```diff
@@ -516,45 +565,126 @@ func (db *DB) FindMigrations() ([]Migration, error) {
 
 // Rollback rolls back the most recent migration
 func (db *DB) Rollback() error {
+	migrations, err := db.FindMigrations()
+	if err != nil {
+		return err
+	}
+	var latest *Migration
+	for _, m := range migrations {
+		if m.Applied {
+			latest = &m
+		}
+	}
+	if latest == nil {
+		return ErrNoRollback
+	}
+	return db.RollbackOnly(migrations, latest.Version)
+}
+
+// MigrateOnly applies exactly one pending migration that matches the given version.
+func (db *DB) MigrateOnly(migrations []Migration, version string) error {
 	drv, err := db.Driver()
 	if err != nil {
 		return err
 	}
 
+	var target *Migration
+	highestApplied := ""
+	for i, m := range migrations {
+		if m.Version == version {
+			target = &migrations[i]
```

Comment:
I just noticed, in some loops you are using `&m` (like line 575) and in some `&migrations[i]`. Maybe this is something you want to align for code quality/consistency.